src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -33,21 +36,58 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     UserService userService = UserServiceFactory.getUserService();
-
+    
+    // Set default variables to create UserAuthenticationData object.
     Boolean loggedIn = false;
+    String email = "";
+    Boolean isUserRegistered = false;
     String authenticationUrl = "";
-    String userEmail = "";
     String redirectUrl = "/";
+
+    // If user is logged in, udpate variables.
+    // Set authenticationUrl to either logout or login URL.
     if (userService.isUserLoggedIn()) {
       loggedIn = true;
-      userEmail = userService.getCurrentUser().getEmail();
+      email = userService.getCurrentUser().getEmail();
       authenticationUrl = userService.createLogoutURL(redirectUrl);
+
+      // Set up variables needed to connect to MySQL database.
+      String url = "jdbc:mysql://localhost:3306/Mintern?useSSL=false&serverTimezone=PST8PDT";
+      String user = "root";
+      String password = "";
```

Comment:
These should either be top level constants, or you might be able to get them from the Utils class where you previously defined these. It's not ideal to have the same values repeated in multiple places, because if you ever want to update any of these values it becomes very error-prone if someone forgets to update every location.

Reply:
Done! (Changed them to the Utility constants.